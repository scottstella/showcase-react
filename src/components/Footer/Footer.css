.footer {
  text-align: center;
  vertical-align: top;
  background-color: #777;
  width: 100%;
}
