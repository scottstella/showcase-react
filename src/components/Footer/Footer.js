import React from "react";

export default function Footer() {
  return (
    <footer>
      <p>Copyright 2022 stella.ws</p>
    </footer>
  );
}
