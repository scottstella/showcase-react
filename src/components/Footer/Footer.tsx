import React from "react";
import "./Footer.css";

const Footer: React.FC = () => {
  return (
    <div className="footer">
      <p>Copyright 2024 stella.ws</p>
    </div>
  );
};

export default Footer;
