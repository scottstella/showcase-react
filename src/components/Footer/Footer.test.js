import React from "react";
import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import Footer from "./Footer";
describe("Footer", () => {
  it("renders without crashing", () => {
    render(React.createElement(Footer, null));
    expect(screen.getByText(/Copyright 2025 stella.ws/i)).toBeInTheDocument();
  });
  it("has the correct CSS class", () => {
    const { container } = render(React.createElement(Footer, null));
    expect(container.firstChild).toHaveClass("footer");
  });
  it("matches snapshot", () => {
    const { container } = render(React.createElement(Footer, null));
    expect(container.firstChild).toMatchSnapshot();
  });
});
