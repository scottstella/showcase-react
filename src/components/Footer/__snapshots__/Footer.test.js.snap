// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > matches snapshot 1`] = `
<div
  class="footer"
>
  <p>
    Copyright 2025 stella.ws
  </p>
</div>
`;
